src/templates/console-v2.html

```diff
@@ -233,6 +239,92 @@
           refreshLine();
         }
 
+        function searchHistory(query, startIndex = -1) {
+          // Search backwards through history from startIndex
+          if (startIndex === -1) {
+            startIndex = history.length - 1;
+          }
+          for (let i = startIndex; i >= 0; i--) {
+            if (history[i].includes(query)) {
+              return i;
+            }
+          }
+          return -1;
+        }
+
+        // reverse-search
+        function enterReverseSearchMode() {
+          reverseSearchMode = true;
+          reverseSearchQuery = "";
+          reverseSearchIndex = -1;
+          savedBuffer = buffer;
+          buffer = "";
+          cursorIndex = 0;
+          updateReverseSearchDisplay();
+        }
+
+        function exitReverseSearchMode(accept = true) {
+          reverseSearchMode = false;
+          if (!accept) {
+            buffer = savedBuffer;
+            cursorIndex = buffer.length;
+          }
+          reverseSearchQuery = "";
+          reverseSearchIndex = -1;
+          savedBuffer = "";
+          refreshLine();
+        }
+
+        function applyReverseSearchSelection() {
+          if (reverseSearchIndex >= 0) {
+            buffer = history[reverseSearchIndex];
+            cursorIndex = buffer.length;
+          }
+        }
+
+        function updateReverseSearchDisplay() {
+          const searchPrompt = `(reverse-i-search)\`${reverseSearchQuery}': `;
+          let displayText = "";
+          if (reverseSearchIndex >= 0) {
+            displayText = history[reverseSearchIndex];
+          }
+          // Clear line and write search prompt with matched command
+          term.write(`\r\x1b[K${searchPrompt}${displayText}`);
+        }
+
+        function reverseSearchAddChar(char) {
+          reverseSearchQuery += char;
+          reverseSearchIndex = searchHistory(reverseSearchQuery);
```

Comment:
The original console does not add a character if there is no matching history. But this will always add the charactor to the query.